pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -1342,16 +1343,35 @@ protected void internalGetPartitionedStats(AsyncResponse asyncResponse, boolean
                 return;
             }
             PartitionedTopicStatsImpl stats = new PartitionedTopicStatsImpl(partitionMetadata);
-            List<CompletableFuture<TopicStats>> topicStatsFutureList = Lists.newArrayList();
+            List<CompletableFuture<TopicStats>> topicStatsFutureList = new ArrayList<>(partitionMetadata.partitions);
+            AtomicBoolean exceptionFlag = new AtomicBoolean(false);
```

Comment:
Maybe we can remove this line and check `exception` at line 1378

Reply:
This check will break the loop if  throws `PulsarServerException`, that will avoid querying the left partitions stats. So I think it is useful @Technoboy-

Reply:
`PulsarServerException` throw by `pulsar().getAdminClient()`. If this occurs, there will be a fast failed result.

Reply:
Nice catch. I get your point.  Have removed the `exceptionFlag`